src/index.ts

```diff
@@ -73,86 +73,100 @@ declare global {
+       */
       preparedStatements?: boolean;
       /**
-       * The database schema in which Graphile Worker is (to be) located.
+       * The database schema in which Graphile Worker's tables, functions, views, etc are
+       * located. Database migrations will create or edit things in this schema
+       * if necessary.
        *
        * @defaultValue `graphile_worker`
        */
       schema?: string;
       /**
-       * Override path to find tasks
+       * The path to a directory in which Graphile Worker should look for task executors.
        *
        * @defaultValue `process.cwd() + "/tasks"`
        */
       taskDirectory?: string;
       /**
-       * Override path to crontab file.
+       * The path to a file in which Graphile Worker should look for crontab schedules.
        *
        * @defaultValue `process.cwd() + "/crontab"`
        */
       crontabFile?: string;
       /**
-       * Number of jobs to run concurrently.
+       * Number of jobs to run concurrently on a single worker.
        *
        * @defaultValue `1`
        */
       concurrentJobs?: number;
-
       /**
-       * A list of file extensions (in priority order) that Graphile Worker
-       * should attempt to import directly when loading tasks. Defaults to
-       * `[".js", ".cjs", ".mjs"]`.
+       * A list of file extensions (in priority order) that Graphile Worker should
+       * attempt to import directly when loading task executors from the file system.
+       *
+       * @defaultValue `[".js", ".cjs", ".mjs"]`
        */
       fileExtensions?: string[];
-
       /**
        * How long in milliseconds after a gracefulShutdown is triggered should
-       * we wait to trigger the AbortController, which should cancel supported
-       * asynchronous actions?
+       * Graphile Worker wait to trigger the AbortController, which should cancel
+       * supported asynchronous actions?
        *
-       * @defaultValue `5000`
+       * @defaultValue `5_000`
        */
       gracefulShutdownAbortTimeout?: number;
-
       /**
-       * Set `true` to use the time as recorded by Node.js rather than
+       * Set to `true` to use the time as recorded by Node.js rather than
        * PostgreSQL. It's strongly recommended that you ensure the Node.js and
        * PostgreSQL times are synchronized, making this setting moot.
+       *
+       * @defaultValue `false`
        */
       useNodeTime?: boolean;
-
       /**
        * **Experimental**
        *
-       * How often should we scan for jobs that have been locked too long and
-       * release them? This is the minimum interval, we'll choose a time between
-       * this and `maxResetLockedInterval`.
+       * How often should Graphile Worker scan for and release jobs that have been
+       * locked too long? This is the minimum interval in milliseconds. Graphile
+       * Worker will choose a time between this and `maxResetLockedInterval`.
+       *
+       * @defaultValue `480_000`
```

Comment:
Keeping this correct is a maintenance burden, do we really need it? Can we provide it another way such that there's only one source of truth?

Reply:
Currently Graphile Config doesn't even know about the subprojects that use it - there's no "registry" or similar - but I don't think we'll keep it isolated like that long term... I think we'll need to know how to merge certain options (should we: overwrite, concatenate, throw, etc?). So maybe if we do that codegen might make sense.

For now, I'm okay with people looking up the defaults in the source and/or docs. Or putting it in the prose to indicate that it might have changed but it was this at one point.

Or just expecting the community to send PRs when they notice the comments are out of sync :wink:

Reply:
By "one place", I meant that each setting's default should be defined in one place, not necessarily one registry to rule them all.

But, if codegen is on the table in the future, then maybe there is a world where graphile config does have a preset registry. That could be interesting.

But, I don't know if I love having the preset *definition* own the default 🤔 It seems like the one of the points of presets is you can somewhat easily share different configurations / different defaults, and customize from there.

Side note, it would be pretty nice to encode more validation than just the TS type in the preset definition. I'm thinking about how one maxPoolSize has to be at least 2, for example. Maybe an opportunity to incorporate a zod schema for a preset definition.

Reply:
All good ideas, but I’m going to delay thinking about them until v6 ;)